internal/controller/api/v1alpha1/nmstate_config_types.go

```diff
@@ -20,19 +20,39 @@ import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
-type RawConfig []byte
+type Interface struct {
+	// nic name used in the yaml, which relates 1:1 to the mac address.
+	// Name in REST API: logicalNICName
+	Name string `json:"name"`
+	// mac address present on the host.
+	// +kubebuilder:validation:Pattern=`^([0-9A-Fa-f]{2}[:]){5}([0-9A-Fa-f]{2})$`
+	MacAddress string `json:"macAddress"`
+}
+
+type RawNetConfig []byte
+
+// NetConfig contains the namestatectl yaml [1] as string instead of golang struct
+// so we don't need to be in sync with the schema.
+//
+// [1] https://github.com/nmstate/nmstate/blob/base/libnmstate/schemas/operational-state.yaml
+// +kubebuilder:validation:Type=object
+type NetConfig struct {
+	Raw RawNetConfig `json:"-"`
```

Comment:
Please add a comment explaining that we have a specific implementation for the unmarshal

Reply:
done